test/xpu_api/random/statistics_tests/philox_uniform_real_distribution_test.pass.cpp

```diff
@@ -0,0 +1,597 @@
+        err += tests_set<sycl::vec<double, 16>, sycl::vec<std::uint_fast64_t, 3>, philox2x64>(queue, 160);
+        err += tests_set<sycl::vec<double, 16>, sycl::vec<std::uint_fast64_t, 2>, philox2x64>(queue, 160);
+        err += tests_set<sycl::vec<double, 16>, sycl::vec<std::uint_fast64_t, 1>, philox2x64>(queue, 160);
+        err += tests_set_portion<sycl::vec<double, 16>, std::uint_fast64_t, philox2x64>(queue, 160, 1);
+        err += tests_set_portion<sycl::vec<double, 16>, std::uint_fast64_t, philox2x64>(queue, 140, 7);
+        err += tests_set_portion<sycl::vec<double, 16>, std::uint_fast64_t, philox2x64>(queue, 160, 17);
+        err += tests_set_portion<sycl::vec<double, 16>, sycl::vec<std::uint_fast64_t, 16>, philox2x64>(queue, 160, 1);
+        err += tests_set_portion<sycl::vec<double, 16>, sycl::vec<std::uint_fast64_t, 8>, philox2x64>(queue, 160, 1);
+        err += tests_set_portion<sycl::vec<double, 16>, sycl::vec<std::uint_fast64_t, 4>, philox2x64>(queue, 160, 1);
+        err += tests_set_portion<sycl::vec<double, 16>, sycl::vec<std::uint_fast64_t, 3>, philox2x64>(queue, 160, 1);
+        err += tests_set_portion<sycl::vec<double, 16>, sycl::vec<std::uint_fast64_t, 2>, philox2x64>(queue, 160, 1);
+        err += tests_set_portion<sycl::vec<double, 16>, sycl::vec<std::uint_fast64_t, 1>, philox2x64>(queue, 160, 1);
+        err += tests_set_portion<sycl::vec<double, 16>, sycl::vec<std::uint_fast64_t, 16>, philox2x64>(queue, 140, 7);
+        err += tests_set_portion<sycl::vec<double, 16>, sycl::vec<std::uint_fast64_t, 8>, philox2x64>(queue, 140, 7);
+        err += tests_set_portion<sycl::vec<double, 16>, sycl::vec<std::uint_fast64_t, 4>, philox2x64>(queue, 140, 7);
+        err += tests_set_portion<sycl::vec<double, 16>, sycl::vec<std::uint_fast64_t, 3>, philox2x64>(queue, 140, 7);
+        err += tests_set_portion<sycl::vec<double, 16>, sycl::vec<std::uint_fast64_t, 2>, philox2x64>(queue, 140, 7);
+        err += tests_set_portion<sycl::vec<double, 16>, sycl::vec<std::uint_fast64_t, 1>, philox2x64>(queue, 140, 7);
+        err += tests_set_portion<sycl::vec<double, 16>, sycl::vec<std::uint_fast64_t, 16>, philox2x64>(queue, 160, 17);
+        err += tests_set_portion<sycl::vec<double, 16>, sycl::vec<std::uint_fast64_t, 8>, philox2x64>(queue, 160, 17);
+        err += tests_set_portion<sycl::vec<double, 16>, sycl::vec<std::uint_fast64_t, 4>, philox2x64>(queue, 160, 17);
+        err += tests_set_portion<sycl::vec<double, 16>, sycl::vec<std::uint_fast64_t, 3>, philox2x64>(queue, 160, 17);
+        err += tests_set_portion<sycl::vec<double, 16>, sycl::vec<std::uint_fast64_t, 2>, philox2x64>(queue, 160, 17);
+        err += tests_set_portion<sycl::vec<double, 16>, sycl::vec<std::uint_fast64_t, 1>, philox2x64>(queue, 160, 17);
+#    endif // TEST_LONG_RUN
+        EXPECT_TRUE(!err, "Test FAILED");
+    }
+
+#endif // TEST_UNNAMED_LAMBDAS
+
+    return TestUtils::done(TEST_UNNAMED_LAMBDAS);
+}
```

Comment:
Please, add a separate test (can be host-side only) with thorough discard check:
generate - then discard and discard after generate + overflow check, you can use base from: https://github.com/rarutyun/iso_cpp_papers/pull/5/files

Reply:
I've extended `philox_test.pass` with this corner-cases testing - https://github.com/oneapi-src/oneDPL/blob/ccc0f1fcb2db03af15f0f87a0d1be2dee45c69c1/test/xpu_api/random/conformance_tests/philox_test.pass.cpp 

@aelizaro , could you please take a look and let me know if that looks fine?